postprocessing/comstockpostproc/utils/hpc.py

```diff
@@ -654,43 +655,121 @@ def count_job_errs(job_err_path, errs_dir):
                 print(f'    {w}')
 
 
-def summarize_failures(yml_path):
+def summarize_failures(yml_path, sort_order='upgrade'):
     """Summarize failures across an whole ComStock run.
 
     This function extracts the ERROR messages for failed runs from the run.log files.
 
     :param yml_path: The path to the YML file used to run buildstockbatch
 
+    :param sort_order: Either 'upgrade' or 'building' to sort the failures by upgrade id first or building id first
+
     :return: None
     """
-    # Extract errors from tar.gz, run in parallel
-    def extract_errors_from_tar(tar_path, fails_dir):
+    # function to merge nested dicts
+    def merge_error_dicts(*dict_args):
+        merged_dict = {}
+        for dict in dict_args:
+            for k, v in dict.items():
+                if k not in merged_dict:
+                    merged_dict[k] = {}
+                merged_dict[k].update(v)
+        return merged_dict
+
+    # Function to sort and write the dictionary based on the option
+    def write_sorted_dict(nested_dict, sort_order, file_name):
+        with open(file_name, 'w') as f:
```

Comment:
Actually we could just flat the nested_dict out into a list = [(key_1, key_2,value), (key_1, key_2, value_2)] and sort it with lambda then print them all.

like:
in the case of sort_order == "building",
[("a", -1, val1),("a", -2, val1), ("b", -2, val2)].sort(key=lambda x : (x[1], x[0]) which will let the list become `[('a', -2, 'k'), ('b', -2, 's'), ('a', -1, 'c')]` which means the x[1] has higher priority in sorting.
then we could skip the repeating flatting logic and and error-prune nested dict filping... IMO.